packages/agreement-process/test/agreementService.integration.test.ts

```diff
@@ -80,145 +91,145 @@ import {
-    startedMongodbContainer = await mongoDBContainer(config).start();
-
-    config.eventStoreDbPort = startedPostgreSqlContainer.getMappedPort(
-      TEST_POSTGRES_DB_PORT
-    );
-    config.readModelDbPort =
-      startedMongodbContainer.getMappedPort(TEST_MONGO_DB_PORT);
-
-    const readModelRepository = ReadModelRepository.init(config);
-    agreements = readModelRepository.agreements;
-    eservices = readModelRepository.eservices;
-    tenants = readModelRepository.tenants;
-
-    readModelService = readModelServiceBuilder(readModelRepository);
-    const eserviceQuery = eserviceQueryBuilder(readModelService);
-    const agreementQuery = agreementQueryBuilder(readModelService);
-    const tenantQuery = tenantQueryBuilder(readModelService);
-    const attributeQuery = attributeQueryBuilder(readModelService);
-
-    postgresDB = initDB({
-      username: config.eventStoreDbUsername,
-      password: config.eventStoreDbPassword,
-      host: config.eventStoreDbHost,
-      port: config.eventStoreDbPort,
-      database: config.eventStoreDbName,
-      schema: config.eventStoreDbSchema,
-      useSSL: config.eventStoreDbUseSSL,
-    });
+let agreements: AgreementCollection;
+let eservices: EServiceCollection;
+let tenants: TenantCollection;
+let readModelService;
+let agreementService: AgreementService;
+let postgresDB: IDatabase<unknown>;
+let startedPostgreSqlContainer: StartedTestContainer;
+let startedMongodbContainer: StartedTestContainer;
+let fileManager: FileManager;
+
+/**
+ * Executes the generic agreement expectation for agreement creation process,
+ * and return the created AgreementV1 object to be used for further checks.
+ *
+ * @param agreementId - The ID of the agreement.
+ * @param expectedEserviceId - The expected e-service ID of the agreement.
+ * @param expectedDescriptorId - The expected descriptor ID of the agreement.
+ * @param expectedProducerId - The expected producer ID of the agreement.
+ * @param expectedConsumerId - The expected consumer ID of the agreement.
+ * @returns A Promise that resolves return the created AgreementV1 object.
+ */
+const expectedAgreementCreation = async (
+  agreementId: AgreementId | undefined,
+  expectedEserviceId: EServiceId,
+  expectedDescriptorId: DescriptorId,
+  expectedProducerId: TenantId,
+  expectedConsumerId: TenantId
+): Promise<AgreementV1> => {
+  expect(agreementId).toBeDefined();
+  if (!agreementId) {
+    fail("Unhandled error: returned agreementId is undefined");
+  }
+
+  const actualAgreementData: StoredEvent | undefined =
+    await readLastAgreementEvent(agreementId, postgresDB);
+
+  if (!actualAgreementData) {
+    fail("Creation fails: agreement not found in event-store");
+  }
+
+  expect(actualAgreementData).toBeDefined();
+  expect(actualAgreementData.type).toBe("AgreementAdded");
+  expect(actualAgreementData.event_version).toBe(1);
+  expect(actualAgreementData.version).toBe("0");
+  expect(actualAgreementData.stream_id).toEqual(agreementId);
```

Comment:
```suggestion
  expect(actualAgreementData).toMatchObject({
    type: "AgreementAdded",
    event_version: 1,
    version: "0",
    stream_id: agreementId,
  });
```

Using `toMatchObject` here gives us a cleaner view on what we expect